src/mobile/src/components/Banner/BannerText.jsx

```diff
@@ -25,19 +25,38 @@ const styles = StyleSheet.create({
   },
 });
 
+async function fetchStudentQuotes() {
+  const { data, error } = await supabase.from('quotes').select('*');
+
+  if (error) {
+    throw new Error(error.message);
+  }
+
+  return data;
+}
+
 const BannerText = () => {
-  const [studentQuote, setStudentQuote] = useState(quotes[0]);
+  const [studentQuotes, setStudentQuotes] = useState({});
 
   useEffect(() => {
-    setStudentQuote(quotes[Math.floor(Math.random() * quotes.length)]);
+    fetchStudentQuotes()
+      .then((quotes) => {
+        const randomIndex = Math.floor(Math.random() * quotes.length);
+        setStudentQuotes(quotes[randomIndex]);
+        return quotes;
```

Comment:
I don't think we have to return quotes since the <code>Text</code> will use <code>studentQuotes</code>

Reply:
When I use `then` it ask me to return something

Reply:
Or you can replace by `await` to do that

Reply:
we can not do that inside `useEffect` as far as I know

Reply:
I think that possible if you wrap you await into an async that call itself 
```js
useEffect(() => {
   (async () => { await ...... })();
});
```

Reply:
I think you did immediate involed async function where is the same as calling `fetchStudentQuotes` ?

Reply:
```js
 const getStudentQuotes = async () {await ..... }
getStudentQuotes.then() // which ask for return by the end ?
```

Reply:
```js
useEffect(() => {
    (async ( ) => { 
        await getStudentQuotes() ;
    })();
});
```

Reply:
@DukeManh @humphd should we give it a go ?

Reply:
Let's not focus on this, leaving it like it is ok. useEffect hook is not an async function. If we write:
```js
 (async ( ) => { 
        await getStudentQuotes() ;
 })();
```
it is making an unnecessary immediately invoked function, which is missing error handling, the current code is good.

We want a `return` here is because of this rule https://eslint.org/docs/rules/consistent-return

Reply:
I see